db/dcrsqlite/sync.go

```diff
@@ -189,30 +202,58 @@ func (db *wiredDB) resyncDB(quit chan struct{}, blockGetter rpcutils.BlockGetter
 			db.waitChan = blockGetter.WaitForHeight(i + 1)
 		}
 
-		if i > stakeDBHeight {
-			if i != int64(db.sDB.Height()+1) {
-				panic(fmt.Sprintf("about to connect the wrong block: %d, %d", i, db.sDB.Height()))
-			}
+		stakeDBHeight = int64(db.sDB.Height())
+		blockHeight := block.Height()
+
+		// A block whose height is less than or equal the stakedb height should
+		// be ignored. Only blocks greater than the stakedb height should be
+		// considered.
+		if blockHeight > stakeDBHeight {
 			if err = db.sDB.ConnectBlock(block); err != nil {
 				return i - 1, err
 			}
 		}
 
+		// If the current block's height fetched earlier, using the notification
+		// waitChan is not greater than the current stakedb height, bypass the waitChan
+		// to fetch the blocks directly till we find a block whose height is greater
+		// than the stakedb height. For blocks whose height is less than the stakedb
+		// height, they exists in the other dbs so skip to the next iteration.
```

Comment:
Its mostly the other way round, like stakedb is behind the wireDB because wireDb is never rewound but stakedb is always rewound to a height matching the auxiliary db. The sync now allows the stakedb to catchup with the wireDB.

Reply:
Ah right the wiredDB will be ahead of both.